### PR TITLE
Update to automatically pick the right layout.

### PR DESCRIPTION
**Pull request by garthwebb**
Also add some logging around MIDI interactions

### CODE REVIEW

File: floor/config/layouts/one-pod-layout.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "squares": 64,
+  "squares": 8,
   "rows": 8,
   "cols": 8,
```
Comment (garthwebb) [inline floor/config/layouts/one-pod-layout.json:4]: "squares" is now meaningful, but rows and cols still aren't (everything still assumes an 8x8 square layout).

File: floor/floor/controller/midi/manager.py [hunk 2]
```diff
@@ -88,11 +88,15 @@ def on_midi_commands(self, peer, commands):
         for command in commands:
             func = mapping.get_function(command)
             if not func:
+                self.logger.info("Received unmapped command: {} {}".format(command[0], command[1]))
```
Comment (garthwebb) [inline floor/floor/controller/midi/manager.py:91]: I spent 30m trying to figure out why my AKAI wasn't working in dev mode before I realized I had not added a "--midi_mapping" flag to give the mapping.  At least this way we'll have some indication if something is trying to talk and we don't understand it.

File: floor/floor/controller/midi/manager.py [hunk 3]
```diff
@@ -88,11 +88,15 @@ def on_midi_commands(self, peer, commands):
...
                 continue
 
             value = command[2]
 
-            func.callback(self.controller, value)
+            try:
+                func.callback(self.controller, value)
+            except AttributeError as e:
```
Comment (garthwebb) [inline floor/floor/controller/midi/manager.py]: I noticed that pressing a mapped button that was not handled by the controller killed everything.  Given that I hope people will be mashing on this APCmini, I don't want random button presses to crash the floor.

File: floor/floor/processor/constants.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 # Maximum value for any component color in a pixel. As currently defined,
 # colors have a 10-bit range of `[0, 1024]`
-COLOR_MAXIMUM = 1024
+COLOR_MAXIMUM = 1023
```
Comment (garthwebb) [inline floor/floor/processor/constants.py:8]: This threw me for a while.  Full brightness patterns were clipping to zero on output and producing a blank floor.

File: floor/floor/processor/simple_step.py [hunk 5]
```diff
@@ -26,20 +24,13 @@ class SimpleStep(Base):
...
 
-    @clocked(frames_per_second=24)
     def get_next_frame(self, context):
         weights = context.weights
         pixels = [(0, 0, 0)] * 64
 
         for idx in range(64):
             if weights[idx] > 0:
-                pixels[idx] = self.hsv_to_rgb([self.hue, self.saturation, self.value])
+                pixels[idx] = self.hsv_to_rgb([self.HUE, self.SAT, self.VAL])
```
Comment (garthwebb) [inline floor/floor/processor/simple_step.py:34]: This is now routed correctly and can be controlled via midi.  This is probably a good overlay for letting people see steps easily.

File: floor/floor/controller/midi/manager.py (hunk 3 above)
Comment (mik3y) [reply floor/floor/controller/midi/manager.py]: got a stack trace for this? feels like root issue is deeper in the stack.

File: floor/floor/processor/constants.py (hunk 4 above)
Comment (mik3y) [reply floor/floor/processor/constants.py:8]: yikes! good catch. can you update the comment above as well.

**Comment by mik3y** [comment]
looks like the change needs to be rebased against master;

```
git fetch
git rebase origin/master
git push -f
```

also looks like travis build is failing for some reason.

File: floor/floor/controller/midi/manager.py (hunk 3 above)
Comment (garthwebb) [reply floor/floor/controller/midi/manager.py]: It was pretty straightforward.  Wrong function used in the mapping file.  Made an update here.

File: floor/floor/processor/constants.py (hunk 4 above)
Comment (garthwebb) [reply floor/floor/processor/constants.py:8]: Updated

File: floor/floor/controller/midi/functions.py [hunk 6]
```diff
@@ -33,7 +33,7 @@ def add(cls, name, **kwargs):
                   callback=lambda controller, _: controller.playlist.previous(),
                   help_text='Go to previous item in the current playlist.')
 MidiFunctions.add('playlist_play',
-                  callback=lambda controller, _: controller.playlist.play(),
+                  callback=lambda controller, _: controller.playlist.start_playlist(),
```
Comment (garthwebb) [inline floor/floor/controller/midi/functions.py:36]: Here's the updated function name.  Did a quick check of the rest as well.

File: floor/floor/controller/midi/manager.py (hunk 3 above)
Comment (mik3y) [reply floor/floor/controller/midi/manager.py]: Ah, in that case, let's catch `Exception`; after all, a future (mis-)implementation could trigger a different kind of exception and we'd need the same protection.

Also, `logger.exception` from within an exception scope will automatically include the stack trace in the log message.

```py
try:
  func.callback(self.controller, value)
except Exception:
  self.logger.exception('Error in MIDI command callback')
```

**Review by mik3y — DISMISSED** [review]
lg!

**Comment by mik3y** [comment]
oh, please un-break the unittests before merging!

in a couple places, it looks like you'll want to update the implementation to use integral/floor division, as we seem to be getting float results.

```
>>> 3/2
1.5
>>> 3//2
1
```

File: floor/floor/controller/midi/manager.py (hunk 3 above)
Comment (garthwebb) [reply floor/floor/controller/midi/manager.py]: Yah, I started with `Exception` and then PyCharm complained that it was too general, and then I googled to see if py-people still felt that a general catch was evil and Python is pure and each exception type is precious and should be handled like a gift, and found it still to be true and I yelled at the computer screen about it and then changed it to a more specific type anyway.

Maybe I'll change it back.

**Review by mik3y — APPROVED** [review]

**Pull request merged**